Tests/NIOEmbeddedTests/EmbeddedChannelTest.swift

```diff
@@ -481,13 +481,22 @@ class EmbeddedChannelTest: XCTestCase {
         XCTAssertEqual(invocations, 1)
     }
 
-    func testSyncOptionsAreSupported() throws {
+    func testGetChannelOptionAutoReadIsSupported() {
         let channel = EmbeddedChannel()
         let options = channel.syncOptions
         XCTAssertNotNil(options)
         // Unconditionally returns true.
         XCTAssertEqual(try options?.getOption(ChannelOptions.autoRead), true)
-        // (Setting options isn't supported.)
+    }
+
+    func testSetGetChannelOptionAllowRemoteHalfClosueIsSupported() {
+        let channel = EmbeddedChannel()
+        let options = channel.syncOptions
+        XCTAssertNotNil(options)
+        // allowRemoteHalfClosure should be false by default
+        XCTAssertEqual(try options?.getOption(ChannelOptions.allowRemoteHalfClosure), false)
+        XCTAssertNoThrow(try options?.setOption(ChannelOptions.allowRemoteHalfClosure, value: true))
+        XCTAssertEqual(try options?.getOption(ChannelOptions.allowRemoteHalfClosure), true)
```

Comment:
Can you extend this test to add a case for changing the value of `allowRemoteHalfClosure` on the `EmbeddedChannel` directly?